storage/cleanup.go

```diff
@@ -117,6 +122,19 @@ func (m *cleanupManager) Report() {
 	}
 }
 
+func (m *cleanupManager) deleteInactiveFilesetFiles() error {
+	multiErr := xerrors.NewMultiError()
+	namespaces, err := m.database.GetOwnedNamespaces()
+	if err != nil {
+		return err
+	}
+	for _, n := range namespaces {
```

Comment:
I think you can localise the code a bit more. For e.g. I don't think you need any `namespace.go` changes if you do the following: 

- add `deleteInactiveFilesetsFn` to the cleanup manager instead of namespace, 
- rewrite the `deleteInactiveFilesetFiles()` function to retrieve the shards from the database namespace, something like:

```
func (m *cleanupManager) deleteInactiveFilesetFiles() error {
	namespaces, err := m.database.GetOwnedNamespaces()
 	if err != nil {
		return err
	}
	multiErr := xerrors.NewMultiError()
	for _, n := range namespaces {
            id := n.ID()
            shards := n.Shards()
            multiErr = multiErr.Add(m.deleteInactiveFilesetsFn(m.filePathPrefix, id, shards)))
        }
        return multiErr.FinalError()
}
```

Reply:
I think it makes sense for a namespace to handle how its deleting the files, particularly since the cleanup function is called wrt a namespace.

If that's the case then the cleanup manager should handle all forms of deletion, vs being called when a namespace is removed.

Reply:
I'm in two minds about this: 

On the one hand, we already have `namespace.CleanupFileset` and `shard.CleanupFileset` which perform fs operations, so sticking with that way of doing things, the way you have the code makes sense. 

That said, the way the code is now, `databaseNamespace` and `databaseShard` both perform fs operations. Once you add the cleanup for unknown namespaces, `databaseCleanupManager` will also perform fs operations. This feels like bad code structure. It'd be easier to reason about/modify if all fs three operations were performed in the `cleanupManager`. 

What do you think?

Reply:
I can get on board with this, it should make it more central and pragmatic to make changes (can make changes in one place vs across a bunch of places).

Reply:
Alright, I like the idea of normalizing /all/ deletion to the cleanup manager. I'll update this PR with those changes.